docs/XBlocks.md

```diff
@@ -0,0 +1,217 @@
+    
+* `Grade Rubric Help Text` - static (developer-defined) help text instructing students to open Project Navigator
+    Resources view to access group project documetns on grading. Help text contains a link that opens the view 
+    automatically.
+    * No author-customizable settings
+* `Submission` - component that encapsulates single group project submission.
+    * `Display Name` - human-friendly submission name - displayed in Project Navigator Submisssions View
+    * `Submission Description` - a longer human-friendly description of a submission - displayed in Project 
+        Navigator Submisssions View
+    * `Upload ID` - upload identifier - used to identify submission to uploads API. Submissions with same upload ID
+        will be updated simultaneously, so it is best to assign some unique value here. Not displayed anywhere, so
+        does not need to be human-comprehensible (i.e. random strings are good idea)
+* `Teammate Selector` - displays a list of teammates and allows switching between them. Only usable in conjunction with
+    `Team Evaluation` stage and `Review Question` blocks. When teammate is selected, `Team Evaluation` stage downloads 
+    current student's responses to `Review Question` and sets corresponding `Review Question` values. Displays current
+    student progress of reviewing other teammates by displaying completion icons next to each teammate.
+    * No author-customizable settings
+    
+    ![Teammate selector screenshot](/docs/images/stage_components/teammate_selector.png)
+    
+* `Group Selector` - displays a list of groups and allows switching between them. Only usable in conjunction with
+    `Peer Grading` stage and `Review Question` blocks. When group is selected, `Peer Grading` stage downloads 
+    current student's responses to `Review Question` and sets corresponding `Review Question` values. Displays current
+    student progress of reviewing other teammates by displaying completion icons next to each teammate. Allows reviewing
+    group submissions by clicking "View team submissions"
+    * No author-customizable settings
+    
+    ![Group selector screenshot](/docs/images/stage_components/group_selector.png)
+    
+* `Review Question` - represents single review question.
+    * `Question ID` - question identifier. Defaults to automatically generated random string
+    * `Question Text` - question text
+    * `Assessment Question Text` - this setting can be used to override `Question Text` when answers displayed for 
+        evaluation. E.g. `Question Text`: "Was it easy to communicate wit this teammate?" can be overridden with 
+        `Assessment Question Text`: "Was it easy to communicate with you?".
+    * `Question Content` - question content. HTML markup that must contain some HTML input control (`input`, 
+        `select` or `textarea`). The value of the control becomes the response to the question.
```

Comment:
@e-kolpakov I think you can shorten this explanation a bit. How about:

"HTML input control (`input`, `select`, or `textarea`) for entering response to the question."